File: go1_commerce/go1_commerce/report/item_wise_sales_report/item_wise_sales_report.py
# Copyright (c) 2013, Tridots Tech and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
from frappe import _
from frappe.query_builder import DocType, Order

def execute(filters=None):
	columns, data = [], []
	columns = get_columns()
	data = get_data(filters)
	return columns, data
	

def get_columns():
	columns = [_("Order ID") + ":Data:150",]
	columns += [	
		_("Order Date") + ":Date:90",
		_("Item ID") + ":Data:140",
		_("Item Name") + ":Data:180",
		_("Customer ID") + ":Data:120",
		_("Customer Name") + ":Data:120",
		_("Customer Email") + ":Data:180", 
		_("Customer Phone") + ":Data:100",
		_("Price") + ":Currency:140",
		_("Quantity") + ":Int:140"
	]
	return columns

def get_data(filters):
	Orders = DocType('Order')
	OrderItem = DocType('Order Item')

	query = frappe.qb.from_(Orders)
	query = query.inner_join(OrderItem).on(Orders.name == OrderItem.parent)

	conditions = []
	
	

	query = query.select(
		Orders.name,
		Orders.order_date,
		OrderItem.item,
		OrderItem.item_name,
		Orders.customer,
		Orders.customer_name,
		Orders.customer_email,
		Orders.phone,
		OrderItem.price,
		OrderItem.quantity
	).where(
		(Orders.docstatus == 1) &
		(Orders.payment_status == 'Paid')
	)
	if filters.get('from_date'):
		query.where(Orders.order_date >= filters.get('from_date'))
	if filters.get('to_date'):
		query.where(Orders.order_date <= filters.get('to_date'))
	query.groupby(Orders.name).orderby(Orders.name, order=Order.desc)
	data = query.run(as_list=True)
	return data

File: go1_commerce/go1_commerce/report/item_wise_sales_report/item_wise_sales_report.js
// Copyright (c) 2016, Tridots Tech and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Item Wise Sales Report"] = {
	"filters": [
		
		{
			"fieldname": "from_date",
			"fieldtype": "Date",
			"label": __("From Date"),
			"default": frappe.datetime.get_today()
		},
		{
			"fieldname":"to_date",
			"label": __("To Date"),
			"fieldtype": "Date",
			"default": frappe.datetime.get_today()
		},
		{
			"fieldname": "business",
			"fieldtype": "Link",
			"label": __("Business"),
			"options": "Business"
		},
	],
	"onload": function() {
		var business_filter = frappe.query_report.get_filter('business');
		if(frappe.session.user != 'Administrator' && has_common(['Vendor'], frappe.user_roles)){
			business_filter.df.hidden = 1;
			business_filter.refresh();
		}
	}
};
